code/game/objects/items/robot/robot_items.dm

```diff
@@ -371,7 +371,7 @@
 	check_amount()
 
 /obj/item/borg/lollipop/proc/dispense(atom/A, mob/user)
-	if(!candy)
+	if(candy >= 0)

```

Comment:
if candy is greater than or equal to zero, tell the user there are no lollipops left in storage ??